servers/auth/auth_dtaas/auth.py

```diff
@@ -0,0 +1,65 @@
+"""Module with functions for authentication microservice
+Authorize new user, Validate existing sessions"""
+
+from requests_oauthlib import OAuth2Session
+import yaml
+
+user_sessions = {}
+
+def authorization():
+    """Function to register/authorize a new user with Auth microservice"""
+    details_file = open("client.yaml",'r')
+    client = yaml.safe_load(details_file)
+
+    client_id = client['id']
+    client_secret = client['secret']
+    redirect_uri = client['redirect_uri']
+    scope = client['scope']
+
+    auth_base_url = "https://gitlab.com/oauth/authorize"
+    token_url = "https://gitlab.com/oauth/token"
+
+    gitlab_session = OAuth2Session(client_id, scope = scope, redirect_uri = redirect_uri)
+    auth_url, state = gitlab_session.authorization_url(auth_base_url)
+
+    print("Go to following link:\n", auth_url)
```

Comment:
print statements need to be replaced with python standard logger package and the output be redirected to stdout